orchestrator/packages/api/src/controllers/grading-queue-controller.ts

```diff
@@ -175,3 +176,22 @@ export const deleteJob = async (req: Request, res: Response) => {
     }
   }
 };
+
+export const getJobStatus = async (req: Request, res: Response) => {
+  const { key, response_url } = req.body;
+  const jobQueueStatus = await getJobQueueStatus(key, response_url);
+  if (!jobQueueStatus) {
+    return res.json("We could not find the job you're looking for. Please contact a professor or admin.");
+  }
+  const { reservation, numReservationsAhead } = jobQueueStatus;
+  const now = new Date();
+  if (now > reservation.releaseAt) {
+    const minutesUntilRelease = Math.floor((now.getTime() - reservation.releaseAt.getTime()) / 60_000);
```

Comment:
See https://github.com/CodeGrade/hourglass/blob/f93b9348e7f00f1181bf6e1fae0b88e95f122a7a/app/packs/components/workflows/student/exams/show/actions/index.ts#L228-L247 for a better/more robust approach to do this.  It uses the `luxon` package, which is a generally good package to have when dealing with datetimes (and especially with timezones...)  It also focuses on times that have passed already, rather than upcoming times, but it's the same principle.